doc/doxygen/src/creating-modules.md

```diff
@@ -68,13 +68,19 @@ their dependencies.
 Modules outside of RIOTBASE                      {#modules-outside-of-riotbase}
 ===========================
 Modules can be defined outside `RIOTBASE`. In addition to add it to `USEMODULE`
-the user needs to add the module path to `EXTERNAL_MODULE_DIRS`.
+the user needs to add the directory (or directories) containing external modules
+to `EXTERNAL_MODULE_DIRS`.
 
-The external module can optionally define the following files:
+External modules can optionally define the following files:
 * `Makefile.include` file to set global build configuration like `CFLAGS` or add
   API headers include paths to the `USEMODULE_INCLUDES` variable.
 * `Makefile.dep` file to set module dependencies
 
+***NOTE:*** The name of an external module must be unique (both in regard to other
+            external modules, as well to native RIOT modules). Additionally, the
+            directory containing the module must match the module name, e.g.
+            module `foo` must be located in `<PATH_IN_EXTERNAL_MODULE_DIRS>/foo`.
```

Comment:
Having had a closer look at the PR I think this hard requirement is the main change that could potentially break external setups as well as make it harder to host bigger out-of-tree projects. I can think of any very good solutions though... @gdoffe used `sed` in his PR, but performing multiple shell calls is a bit slow.

At the same time `EXTERNAL_MODULE_DIRS` although less convenient actually maps what happens with Modules currently in RIOT regarding this, modules in RIOT are not automatically part of a search path either.

But this is only regarding compilation, `Makefile` inclusion inherently requires what is written in this `note` since it's based on rules such as: `-include $(USEPKG:%=$(RIOTPKG)/%/Makefile.dep)` or `-include $(sort $(USEMODULE:%=$(RIOTBASE)/drivers/%/Makefile.dep))` or `-include $(EXTERNAL_MODULE_DIRS:%=%/Makefile.dep)`, so if the module name does not match the directory name then no `Makefile.%` would get included. (Don't know if somewhere it says that a module with `Makefile.%` must have the same name as its directory for drivers).

So anyway we are already dealing with nondefault behavior to get some convenience, so trading one for another is fine if most agree, let's give you a forum post some time and then maybe add a `Make` warning for a while if there is `EXTERNAL_MODULE_DIRS` usage somewhere.

Reply:
Aren't unique module names already a requirement for RIOT modules?

Reply:
> Aren't unique module names already a requirement for RIOT modules?

For the same build yes, but if there is no overlap, lets say between two different BOARD's, then both can have a module names `periph`. But here the important part is that directories - module would be 1:1, and although in the same build module names must be unique directories don't.

Reply:
> can we have multiple EXTERNAL_MODULE_DIRS though?

Sure :-) the same already applies to EXTERNAL_BOARD_DIRS. I have one Repo with work boards, and one repo with home grown private boards in it :-)